Frontend/ui-driver/src/Helpers/Utils.purs

```diff
@@ -239,12 +239,47 @@ capitalizeFirstChar inputStr =
     in DS.joinWith " " output
 
 getDowngradeOptions :: String -> Array String
-getDowngradeOptions variant = case variant of
-                                "TAXI"  -> []
-                                "SUV"   -> ["SEDAN", "HATCHBACK"]
-                                "SEDAN" -> ["TAXI", "HATCHBACK"] 
-                                _       -> ["TAXI"]
-
+getDowngradeOptions variant = case (getMerchant FunctionCall) of
+                                YATRISATHI -> case variant of
+                                                "TAXI"  -> []
+                                                "SUV"   -> ["SEDAN", "HATCHBACK"]
+                                                "SEDAN" -> ["TAXI", "HATCHBACK"] 
+                                                _       -> ["TAXI"]
+                                _ -> case variant of
+                                        "SUV"   -> ["SEDAN", "HATCHBACK"]
+                                        "SEDAN" -> ["HATCHBACK"]
+                                        _       -> []
+
+
+getDowngradeOptionsText :: String -> String
+getDowngradeOptionsText vehicleType = do
+  let
+    downgradeFrom = case (getMerchant FunctionCall) of
+                    YATRISATHI -> getVariantRideType vehicleType
+                    _ -> getVehicleType vehicleType
+    downgradeTo = case (getMerchant FunctionCall) of
+                    YATRISATHI -> case vehicleType of
+                                    "SUV" -> getString AC_CAB
+                                    _     -> getString TAXI
+                    _ -> case vehicleType of
+                          "SEDAN" -> getString HATCHBACK
+                          _ -> getString TAXI
```

Comment:
why we have case matching here again, instead we should define variant mappings as per type in `getVariantRideType` function only and use that, so that we don't need to change everywhere but only at one place